### PR TITLE
Ensure AnimationPlayer evaluate animations when autoplay is enabled and node becomes ready

### LINKED ISSUE #83326: Autoplay animation of `AnimationPlayer` starts one frame late
### Godot version

v4.2.beta1.official [b1371806a]

### System information

Godot v4.2.beta1 - Windows 10.0.22621 - Vulkan (Forward+) - integrated Intel(R) Iris(R) Xe Graphics (Intel Corporation; 31.0.101.3889) - 12th Gen Intel(R) Core(TM) i7-1255U (12 Threads)

### Issue description

In Godot 4.2, autoplayed animations by `AnimationPlayer` no longer set properties on the frame the node enters the tree. It rather does it one frame later, which can cause visuals to "flash" for a frame if the default values of the properties are different from the ones of the first property keys of the autoplayed animation, which happened in my project when porting it to 4.2.

(Here I set Engine.max_fps to 15 to make the bug clearer)

https://github.com/godotengine/godot/assets/107559799/5972b8fc-1fe6-4f2f-9c3a-4c64e5862c69

### Steps to reproduce

1. Create an `AnimationPlayer` and a node to animate
2. Create a new animation in the newly created `AnimationPlayer`
3. Add a new property track in the animation with RESET keys
4. Set the first key's value to one different from the RESET animation's value
5. Set this animation to autoplay
6. Add a script which prints the property value on ready:
```gdscript
func _ready():
    print($<path to animated node>.<animated property>)
```
7. Run the scene and notice how it prints the default value

### Minimal reproduction project

[animation_autoplay_bug.zip](https://github.com/godotengine/godot/files/12905385/animation_autoplay_bug.zip)

### PR DESCRIPTION
**Pull request by martinfelis**
Fixes #83326.

The issue was introduced in the refactoring done in 1b95827d3ef244de322b0c16deb49fefe48ed1a1.

### CODE REVIEW

File: scene/animation/animation_player.cpp [hunk 1]
```diff
@@ -151,6 +151,7 @@ void AnimationPlayer::_notification(int p_what) {
 			if (!Engine::get_singleton()->is_editor_hint() && animation_set.has(autoplay)) {
 				set_active(true);
 				play(autoplay);
+				_process_animation(0);
```
Comment (TokageItLab) [inline scene/animation/animation_player.cpp:154]:
```suggestion
				seek(0, true);
```
To consist with other line like `_stop_internal()`.

**Review by TokageItLab — CHANGES_REQUESTED** [review]

File: scene/animation/animation_player.cpp (hunk 1 above)
Comment (martinfelis) [reply scene/animation/animation_player.cpp]: Thanks for the suggestion. See ed88521.

**Review by TokageItLab — APPROVED** [review]
LGTM, needs to squash.
https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**